File: app/src/androidTest/java/daotests/DirectMessageDaoTest.java
package daotests;

import org.junit.After;
import org.junit.AfterClass;
import org.junit.Before;
import org.junit.BeforeClass;

import androidx.room.Dao;

public class DirectMessageDaoTest extends DaoTest {


    @BeforeClass
    public static void initDatabase() {

    }


    @After
    public void clearTables() {
        testDatabase.clearAllTables();
    }

    @AfterClass
    public static void closeDatabase() {
        testDatabase.close();
        testDatabase = null;
    }
}

File: app/src/androidTest/java/daotests/DirectMessageDaoTest.kt
package daotests

import android.content.ContentValues
import android.database.Cursor
import android.database.sqlite.SQLiteDatabase
import androidx.room.Dao
import com.klinker.android.twitter_l.data.roomdb.daos.DirectMessageDao
import com.klinker.android.twitter_l.data.roomdb.entities.DirectMessage
import com.klinker.android.twitter_l.data.roomdb.entities.User
import com.klinker.android.twitter_l.data.roomdb.pojos.DisplayDirectMessage
import matchers.DisplayDirectMessageMatcher
import matchers.DisplayDirectMessageMatcher.Companion.matchesDisplayDirectMessage
import org.hamcrest.MatcherAssert.assertThat
import org.hamcrest.Matchers.*
import org.junit.*

class DirectMessageDaoTest : DaoTest()